interop-testing/src/main/java/io/grpc/testing/integration/OrcaTestServiceImpl.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.testing.integration;
+
+import com.google.common.collect.ImmutableMap;
+import io.grpc.ForwardingServerCall;
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.services.CallMetricRecorder;
+import io.grpc.xds.orca.OrcaOobService;
+import java.util.concurrent.ScheduledExecutorService;
+import javax.annotation.Nullable;
+
+public class OrcaTestServiceImpl extends TestServiceImpl {
```

Comment:
This interceptor overall is pretty hacky. It'd be fine if this was a unit test, but it doesn't fit well here. I'd really like us to send in the request proto what cost to include in the response trailers, and have the TestServiceImpl call CallMetricRecorder.